### PR TITLE
List requests to view

### PR DESCRIPTION
**Pull request by amit-mb**
The requests urls and the request method are displayed now. Used MessageService to get the incoming requests and added the requests to state in Popup component which is passed as props to RequestList component which was already mapping the urls to a table. 

As of now, only the request url's and methods are rendered.

### CODE REVIEW

File: app/background.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ const DEFAULT_DATA : Data = {
   count: 0
 };
 
-class BackgroundWorker {
+export class BackgroundWorker {
```
Comment (revathskumar) [inline app/background.ts]: Why we need to export this?

File: app/background.ts [hunk 2]
```diff
@@ -31,6 +31,8 @@ class BackgroundWorker {
     chrome.runtime.onMessage.addListener((request, _, sendResponse) => {
       const tabId = request.tabId;
 
+      //console.log('message service :: ', request);
```
Comment (revathskumar) [inline app/background.ts]: remove log.

File: app/getRequests.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+let requestsArr:Array<Object> = [{url : "", requestBody: undefined , method : "", type: ""}]
```
Comment (revathskumar) [inline app/getRequests.tsx]: This is not using anywhere. so we can remove this.

File: app/popup.tsx [hunk 4]
```diff
@@ -11,18 +13,26 @@ const isChromeUrl = (url : string) => {
...
       label: props.enabled ? 'Stop Listening' : 'Start Listening',
-      errorMessage: ''
+      errorMessage: '',
+      requests: [],
     };
   }
 
+  componentDidMount() {
+    MessageService.getRequests(this.props.tabId, (requests) => {
+      console.log('reqs :: ', requests);
```
Comment (revathskumar) [inline app/popup.tsx]: can remove the logs

File: app/getRequests.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+chrome.webRequest.onBeforeRequest.addListener(
+function(details) {
```
Comment (kgrz) [inline app/getRequests.tsx]: Try spend some time setting up your editor. And try using https://github.com/prettier/prettier

Comment (amit-mb) [reply app/getRequests.tsx]: Sure Kashyap. Will do that and follow .editorconfig

File: app/popup.tsx (hunk 4 above)
Comment (amit-mb) [reply app/popup.tsx]: Will do the changes and push accordingly

File: package-lock.json [hunk 6]
```diff
@@ -0,0 +1,3487 @@
+{
```
Comment (revathskumar) [inline package-lock.json]: please remove the package-lock.json. we already have yarn.lock

**Pull request merged**